baserun/baserun.py

```diff
@@ -20,245 +40,326 @@ class BaserunEvaluationFailedException(Exception):
 
 class Baserun:
     _initialized = False
-    _trace_id = None
-    _traces = []
 
-    _buffer = []
-    _buffer_lock = threading.Lock()
+    _api_key = None
 
-    _evals = []
-    _evals_lock = threading.Lock()
+    _instrumentors: list[BaseInstrumentor] = None
 
-    _api_base_url = None
-    _api_key = None
+    grpc_channel: grpc.Channel = None
+    submission_service: SubmissionServiceStub = None
+
+    current_test_suite: TestSuite = None
 
     evals = Evals
 
     @staticmethod
-    def init(api_base_url: str = "https://baserun.ai/api/v1") -> None:
-        api_key = os.environ.get('BASERUN_API_KEY')
+    def init(instrument: bool = True) -> None:
+        api_key = os.environ.get("BASERUN_API_KEY")
         if not api_key:
-            raise ValueError("Baserun API key is missing. Ensure the BASERUN_API_KEY environment variable is set.")
+            raise ValueError(
+                "Baserun API key is missing. Ensure the BASERUN_API_KEY environment variable is set."
+            )
 
         if Baserun._initialized:
-            warnings.warn("Baserun has already been initialized. Additional calls to init will be ignored.")
             return
 
-        Baserun._api_base_url = api_base_url
         Baserun._api_key = api_key
         Baserun._initialized = True
-        Baserun.evals.init(Baserun._append_to_evals)
 
-        OpenAIWrapper.init(Baserun._handle_auto_llm)
-        AnthropicWrapper.init(Baserun._handle_auto_llm)
+        if key_chain := os.environ.get("SSL_KEY_CHAIN"):
```

Comment:
See what I did for the typescript sdk. Basically creating the submission service at [runtime](https://github.com/baserun-ai/baserun-js/compare/main...apg-otel-js#diff-b9f13ecbedb5624bd013c6cc0bdcd98884c4f0fae8a10224a822b5c78dd55730R6) . Reason for that is twofold:
1. Break circular dependencies so evals, etc don't need to require Baserun
2. Let's people set the api-key late. In this case we heard user feedback that setting baserun.api_key can be nice so we don't have to worry about ordering around init() and environment variable loading

Reply:
Seems reasonable, would like to add that change in later if possible